sdk/test-utils/test-credential/src/index.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ClientSecretCredential } from "@azure/identity";
+import { env, isPlaybackMode } from "@azure-tools/test-recorder";
+import { NoOpCredential } from "./noOpCredential";
+
+/**
+ * ## Credential to be used in the tests.
+ *
+ * ### In playback mode
+ *  - returns the NoOpCredential (helps bypass the AAD traffic)
+ *
+ * ### In record/live modes
+ *  - returns the ClientSecretCredential (expects AZURE_TENANT_ID, AZURE_CLIENT_ID, AZURE_CLIENT_SECRET in your environment or in the .env file)
+ *  - AAD traffic won't be recorded if this credential is used.
+ */
+export function createTestCredential() {
```

Comment:
please take an optional `TokenCredentialOptions` parameter.

Reply:
Added that.

Should I do CSC options instead?
![image](https://user-images.githubusercontent.com/10452642/146610452-72e269e5-6bcf-42ba-961d-1284e7825bbb.png)

Reply:
@sadasant

Reply:
I haven't seen the need to use the persistence option, but @sadasant can advise here.

Reply:
I would leave it as TokenCredentialOptions to avoid guaranteeing that this API uses CSC in the back end. I am optimistic that we can eventually make this work with DefaultAzureCredential, even if we have to plumb something through the karma request pipeline.